packages/next/src/lib/metadata/resolve-metadata.ts

```diff
@@ -82,7 +82,11 @@ function merge(
     openGraph: string | null
   }
 ) {
-  const metadataBase = source?.metadataBase || target.metadataBase
+  // If there's override metadata, prefer it otherwise fallback to the default metadata.
+  const metadataBase =
+    typeof source?.metadataBase !== 'undefined'
+      ? source.metadataBase
+      : source?.metadataBase || target.metadataBase
```

Comment:
```suggestion
      : target.metadataBase
```

The first case of the ternary would return if it was truthy so you dont need to repeat it again here, right?